homeassistant/util/yaml.py

```diff
@@ -119,10 +120,15 @@ def _env_var_yaml(loader, node):
         raise HomeAssistantError(node.value)
 
 
+def _secret_yaml(loader, node):
+    """Load secrets and embed it into the configuration YAML."""
+    return secrets.get_secret(node.value) or ''
```

Comment:
At the moment we can blow up at the end with a list of unresolved secrets.
The problem is when you have 10-20 secrets in keyring, preferably you don't want to run hass and find missing secrets one by one. Just a list when it blows up then would be better I think.

`HISTORY` only keeps the source of the secret in the latest version, not the secret itself (keyring / secret.yaml)

`secrets.yaml` will always be insecure, since it's file based, so keeping that in secrets.SECRET_DICT is not that bad (but can hide this and only expose `get_secret`)